### ISSUE TITLE
[Discussion] Scoring system rework

### ISSUE THREAD

**Issue by frabul** [supermedium/moonrider#194]
I have been looking at the actual scoring system.
Just for fun i have implemented a new system that takes into account

- speed
- angle
- how much of the box was sliced 
- accuracy ( distance of slice line from center )

This system is a lot more challenging, as the scores are much diffent between a good and bad slash so i thought that it could be merged in main repo.
You can try it here https://frabul.github.io/moonrider/?debugstates=hits, it will display some insights on the last slash too. In that version i also reduced respawn distance a lot but that is not related.

What do you think?
If you like that i can clean up and create a merge request.

**Comment by dmarcos** [comment]
Thanks. I recommend sharing it with the community to see how people like it

**Comment by frabul** [comment]
Is still there a discord channel? I can't find it.

**Comment by dmarcos** [comment]
> Is still there a discord channel? I can't find it.

Yes, https://discord.gg/62dtQrr9

**Comment by fliodhais** [comment]
I'm mainly a punch mode player and was abit disappointed to find that it instantly crashes punch mode on first hit. I understand this is just a quick prototype but maybe applying the score only to saber mode would be good.

Otherwise, I tried out saber and it seemed abit too unforgiving to me. I like the idea of being more precise giving more score but I think this would discourage newcomers to the game. I suggest maybe we could 2-3 zones like an archery ring system with actual visual queues so its abit easier to tell. 

Additionally, indirectly related to your feature is that I think a good feature to have would be a more dynamic score counter or something that shows like points and combo streak in a more heads on manner (not too far off periphery view)

**Comment by frabul** [comment]
Thanks for taking the time of trying.

> I'm mainly a punch mode player and was abit disappointed to find that it instantly crashes punch mode on first hit. I understand this is just a quick prototype but maybe applying the score only to saber mode would be good.

You are right, there is a bug for punch mode and i was sloppy to not fix it, mainly because **punch mode is unchanged** as the idea of slicing does not apply to it, if you have any suggestions on how to improve socring for punch mode i can try to modify it. Maybe accuracy could be added to the equation. Of course the bug will be fixed in case of merge.

> Otherwise, I tried out saber and it seemed abit too unforgiving to me. I like the idea of being more precise giving more score but I think this would discourage newcomers to the game. I suggest maybe we could 2-3 zones like an archery ring system with actual visual queues so its abit easier to tell.

I don't completely understand what you are suggesting here. However, the system is open for tuning... 
About unforgiveness
Probably, what is preventing you from successful hit is not the accuracy of the slash but the minimum "slice completion" requirement, that indicates if the blade passed trough the whole box or only some portion of it.
 About unforgiveness, that is probably true: at first i set a minimum slicing of 100% of the box width and it was really difficult and after some tests i decreased to 20% ( like it is right now ). I also thought about the possibility of removing the slice requirement altoghether and leave only the penalty of not taking full score ( but still counting as successful hit ).

**Comment by frabul** [comment]
> Additionally, indirectly related to your feature is that I think a good feature to have would be a more dynamic score counter or something that shows like points and combo streak in a more heads on manner (not too far off periphery view)

Yes, that is on the list. I also stubled upon a file that seems to implement this feature already but for some reason it is disabled ( not working? performance issues ? ). 
Right now i'm trying to understand the reason behind lag spikes, it seems to be related to garbage collector.
